test/functional/testnvim.lua

```diff
@@ -759,58 +759,21 @@ function M.assert_visible(bufnr, visible)
   end
 end
 
---- @param path string
-local function do_rmdir(path)
-  local stat = uv.fs_stat(path)
-  if stat == nil then
-    return
-  end
-  if stat.type ~= 'directory' then
-    error(string.format('rmdir: not a directory: %s', path))
-  end
-  for file in vim.fs.dir(path) do
-    if file ~= '.' and file ~= '..' then
-      local abspath = path .. '/' .. file
-      if t.isdir(abspath) then
-        do_rmdir(abspath) -- recurse
-      else
-        local ret, err = os.remove(abspath)
-        if not ret then
-          if not session then
-            error('os.remove: ' .. err)
-          else
-            -- Try Nvim delete(): it handles `readonly` attribute on Windows,
-            -- and avoids Lua cross-version/platform incompatibilities.
-            if -1 == M.call('delete', abspath) then
```

Comment:
This is effectively just a wrapper to `uv_fs_unlink()`.

Reply:
perhaps not needed anymore now that we have luv (previously were using Lua stdlib or "lfs", which didn't work well for Windows).

but the main problem that was being solved here is when files are still open in another process on Windows, the failure is unclear. we can revisit in `vim.fs.rm` if needed.